### PR TITLE
integrate SQON to URL query params

### PR DESCRIPTION
**Pull request by justincorrigible**

### CODE REVIEW

File: components/pages/explorer/PageContent.tsx [hunk 1]
```diff
@@ -19,14 +19,46 @@
  *
  */
 
+import { useEffect, useState } from 'react';
 import { css } from '@emotion/core';
-import RepoTable from './RepoTable';
+import stringify from 'fast-json-stable-stringify';
+import { isEqual } from 'lodash';
+
+import useUrlParamState from '@/global/hooks/useUrlParamState';
```
Comment (justincorrigible) [inline components/pages/explorer/PageContent.tsx:27]: Brought (a modified version of) this helper from Platform-UI

File: components/pages/explorer/PageContent.tsx [hunk 2]
```diff
@@ -19,14 +19,46 @@
...
 const PageContent = (props: PageContentProps) => {
+  const [firstRender, setFirstRender] = useState<boolean>(true);
+  const [currentFilters, setCurrentFilters] = useUrlParamState<RepoFiltersType | null>(
+    'filters',
+    null,
+    {
+      deSerialize: (v) => (v ? JSON.parse(v) : null),
+      serialize: (v) => (v ? stringify(v) : ''),
+    },
+  );
```
Comment (justincorrigible) [inline components/pages/explorer/PageContent.tsx:49]: invoke the helper here

File: components/pages/explorer/PageContent.tsx [hunk 3]
```diff
@@ -19,14 +19,46 @@
...
+
+  // TODO: abstract these effects into an Arranger integration
+  useEffect(() => {
+    currentFilters && props.setSQON(currentFilters);
+    setFirstRender(false);
+  }, []);
```
Comment (justincorrigible) [inline components/pages/explorer/PageContent.tsx:55]: on first render, grab any query params and use them to hydrate the SQON

File: components/pages/explorer/PageContent.tsx [hunk 4]
```diff
@@ -19,14 +19,46 @@
...
+
+  useEffect(() => {
+    const { sqon } = props;
+    firstRender || isEqual(sqon, currentFilters) || setCurrentFilters(sqon);
+  }, [props.sqon]);
```
Comment (justincorrigible) [inline components/pages/explorer/PageContent.tsx:60]: conversely, whenever the SQON changes (except upon first render), update the URL

File: global/config.ts [hunk 5]
```diff
@@ -62,9 +62,10 @@ export const getConfig: () => AppConfig = () => {
     NEXT_PUBLIC_ADMIN_EMAIL: publicConfig.NEXT_PUBLIC_ADMIN_EMAIL,
     NEXT_PUBLIC_DOMAIN_ROOT_URL:
       publicConfig.NEXT_PUBLIC_DOMAIN_ROOT_URL || 'http://localhost:3000',
+    NEXT_PUBLIC_GOOGLE_ANALYTICS_DEBUG: publicConfig.NEXT_PUBLIC_GOOGLE_ANALYTICS_DEBUG || false,
```
Comment (justincorrigible) [inline global/config.ts]: Bonus: Adding something to (optionally) quiet the GA down a little bit.

File: global/hooks/useUrlParamState.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+): Record<string, string> => {
+  const currentQueryEntries = [...new URLSearchParams(queryString).entries()];
+  return currentQueryEntries.length
+    ? currentQueryEntries.reduce((acc, [key, val]) => ({ ...acc, [key]: val }), {})
+    : {};
+};
+
+const getSanitizedValue = (v: string): null | undefined | string => (v ? v : null);
+
+export const useHook = <T>(
```
Comment (justincorrigible) [inline global/hooks/useUrlParamState.ts:34]: The aforementioned hook from Platform-UI

File: tsconfig.json [hunk 7]
```diff
@@ -3,6 +3,8 @@
     "target": "es5",
     "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
+    "baseUrl": ".",
+    "downlevelIteration": true,
```
Comment (justincorrigible) [inline tsconfig.json:7]: this was necessary for the new hook. 
TypeScript wasn't having URLQueryParams be iterated.

File: tsconfig.json [hunk 8]
```diff
@@ -12,7 +14,13 @@
...
     "resolveJsonModule": true,
     "isolatedModules": true,
-    "jsx": "preserve"
+    "jsx": "preserve",
+    "paths": {
+      "@/components/*": ["components/*"],
+      "@/config": ["global/config.ts"],
+      "@/global/*": ["global/*"],
+      "@/pages/*": ["pages/*"]
+    }
```
Comment (justincorrigible) [inline tsconfig.json:23]: and this last little thing to stop the `../../../../../../../` ... madness

**Review by justincorrigible — COMMENTED** [review]
Quick pointers to help navigate what should be a fairly simple PR 🤞

**Review by anncatton — APPROVED** [review]

**Pull request merged**